### PR TITLE
 JAMES-2622 Introduce env variable for java options in dockerfiles 

### PR DESCRIPTION
**Pull request by aduprat**
Only last commit: 8a4f086

### CODE REVIEW

File: dockerfiles/run/guice/cassandra-ldap/Dockerfile [hunk 1]
```diff
@@ -28,4 +28,6 @@ ADD destination/conf /root/conf
 VOLUME /logs
 VOLUME /root/conf
 
-ENTRYPOINT java -Dlogback.configurationFile=/root/conf/logback.xml -Dworking.directory=/root/ -jar james-server.jar 
+ENV JVM_OPTIONS="-XX:-PrintGC"
```
Comment (mbaechler) [inline dockerfiles/run/guice/cassandra-ldap/Dockerfile]: why did you choose that option by default ?

File: dockerfiles/run/guice/cassandra-ldap/Dockerfile [hunk 2]
```diff
@@ -28,4 +28,6 @@ ADD destination/conf /root/conf
...
+
+ENTRYPOINT java -Dlogback.configurationFile=/root/conf/logback.xml -Dworking.directory=/root/ $JVM_OPTIONS -jar james-server.jar
```
Comment (rouazana) [inline dockerfiles/run/guice/cassandra-ldap/Dockerfile]: this should not work with options containing spaces. Do we care?

Comment (rouazana) [reply dockerfiles/run/guice/cassandra-ldap/Dockerfile]: oh you missed my point, `--env JVM_OPTIONS="-Xms256m -Xmx2048m"` is passing two options `-Xms256m` and `-Xmx2048m`. My point is about having one option containing a space (for example for a path).

Comment (aduprat) [reply dockerfiles/run/guice/cassandra-ldap/Dockerfile]: Sorry, I don't get you...
Can you give me an example of valid jvm options you are thinking of?

Comment (rouazana) [reply dockerfiles/run/guice/cassandra-ldap/Dockerfile]: `-Xloggc:/path with spaces/gc.log`

Comment (aduprat) [reply dockerfiles/run/guice/cassandra-ldap/Dockerfile]: Do we really want to handle such cases?

Comment (rouazana) [reply dockerfiles/run/guice/cassandra-ldap/Dockerfile]: That's why I started by " Do we care?" ;)

Comment (aduprat) [reply dockerfiles/run/guice/cassandra-ldap/Dockerfile]: My opinion is: "no"
But I'm not the PO ;)

Comment (rouazana) [reply dockerfiles/run/guice/cassandra-ldap/Dockerfile]: can you just check if dockerFile handles "${JVM_OPTS}" syntax? (that's a shell hack)

**Comment by trantienduchn** [comment]
test this please

**Comment by rouazana** [comment]
test this please

**Comment by rouazana** [comment]
test this please

**Comment by rouazana** [comment]
test this please

File: dockerfiles/run/guice/cassandra-ldap/Dockerfile (hunk 2 above)
Comment (aduprat) [reply dockerfiles/run/guice/cassandra-ldap/Dockerfile]: Good catch, it's supported in Dockerfiles.

File: dockerfiles/run/guice/cassandra/Dockerfile [hunk 3]
```diff
@@ -28,4 +28,4 @@ ADD destination/conf /root/conf
 VOLUME /logs
 VOLUME /root/conf
 
-ENTRYPOINT java -Dlogback.configurationFile=/root/conf/logback.xml -Dworking.directory=/root/ -jar james-server.jar 
+ENTRYPOINT java -Dlogback.configurationFile=/root/conf/logback.xml -Dworking.directory=/root/ "${JVM_OPTIONS}" -jar james-server.jar
```
Comment (mbaechler) [inline dockerfiles/run/guice/cassandra/Dockerfile]: why it's the only place you wrap the string with quotes ?

**Comment by aduprat** [comment]
> Be careful to not merge alpine image usage, too

Rebased on `master`

File: dockerfiles/run/guice/cassandra/Dockerfile (hunk 3 above)
Comment (aduprat) [reply dockerfiles/run/guice/cassandra/Dockerfile]: Because I need some coffee.

**Comment by aduprat** [comment]
```
[ERROR] Errors: 
[ERROR]   JettyHttpServerTest.shouldStartOnConfiguredPort:106 » Bind Address already in ...
```
test this please

**Comment by aduprat** [comment]
test this please

**Comment by aduprat** [comment]
test this please

**Comment by aduprat** [comment]
test this please

**Comment by aduprat** [comment]
test this please

**Comment by aduprat** [comment]
test this please

**Comment by aduprat** [comment]
test this please

**Comment by aduprat** [comment]
test this please

**Comment by aduprat** [comment]
test this please

**Comment by aduprat** [comment]
test this please

**Comment by aduprat** [comment]
test this please

**Comment by aduprat** [comment]
test this please

**Comment by aduprat** [comment]
test this please

**Comment by aduprat** [comment]
Rebased on `master`

**Comment by aduprat** [comment]
> Rebased on `master`
again

**Comment by aduprat** [comment]
```
Caused by: java.io.IOException: Server returned HTTP response code: 502 for URL: https://api.github.com/repos/linagora/james-project/statuses/c7ae06a286b0915226e762a0d9d26e797fc3c6ab
```

test this please

**Comment by trantienduchn** [comment]
```
...
[c7ae06a286b0915226e762a0d9d26e797fc3c6ab] + docker exec cassandra-james-2622-options-60ea52bd-4818-4d8c-8295-87b4b5046ee4 cqlsh -e describe columnfamily system.local;
[c7ae06a286b0915226e762a0d9d26e797fc3c6ab] Connection error: ('Unable to connect to any servers', {'127.0.0.1': error(111, "Tried connecting to [('127.0.0.1', 9042)]. Last error: Connection refused")})
[c7ae06a286b0915226e762a0d9d26e797fc3c6ab] Running: Shell Script
[c7ae06a286b0915226e762a0d9d26e797fc3c6ab] [248] Running shell script
...
```

**Comment by mbaechler** [comment]
merged

**Pull request closed without merging**